src/main/java/picard/sam/markduplicates/UmiAwareMarkDuplicatesWithMateCigar.java

```diff
@@ -128,19 +130,22 @@
     public boolean ALLOW_MISSING_UMIS = false;
 
     private final Log log = Log.getInstance(UmiAwareMarkDuplicatesWithMateCigar.class);
-    private UmiMetrics metrics = new UmiMetrics();
+    private final Map<String, UmiMetrics> metrics = new HashMap<>();
 
     @Override
     protected int doWork() {
         // Before we do anything, make sure the UMI_METRICS_FILE can be written to.
         IOUtil.assertFileIsWritable(UMI_METRICS_FILE);
 
         // Perform Mark Duplicates work
-        int retval = super.doWork();
+        final int retval = super.doWork();
+
+        // Add results in metrics to the metricsFile
+        final MetricsFile<UmiMetrics, Double> metricsFile = getMetricsFile();
+        for (final Map.Entry<String, UmiMetrics> metric : metrics.entrySet()) {
```

Comment:
As before, using `metrics.values()` would be better here since the entry's key isn't used.